R/plot-build.R

```diff
@@ -84,11 +84,18 @@ ggplot_build.ggplot <- function(plot) {
   layout$setup_panel_params()
   data <- layout$map_position(data)
 
-  # Train and map non-position scales
+  # Hand off position guides to layout
+  layout$setup_panel_guides(plot$guides, plot$layers)
+
+  # Train and map non-position scales and guides
   npscales <- scales$non_position_scales()
   if (npscales$n() > 0) {
     lapply(data, npscales$train_df)
+    plot$guides <- plot$guides$build(npscales, plot$layers, plot$labels)
```

Comment:
So at this point, it is still unclear whether guides should see mapped or unmapped data.
I've opted for unmapped data here, but if it should be mapped, than we need to swap this line with the line below.